### PR TITLE
[master] Move templates from .tpp to .cpp with explicit instantiation

### PR DESCRIPTION
**Pull request by valeryz**
## Description
Most of the change is actually recovering #includes lost because of lack of transitive includes from the removed .tpp

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [ ] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

**Comment by valeryz** [comment]
This is something like what @art-gor suggested.

And I just discussed with @CSideSteve .   

Measured the impact on build times, and the win is marginal.  On my 8-core Intel i7-6700 4 GHz (5 y.o. CPU) it is  7 minutes before with this PR vs. 6 minutes after.

Incremental build time impact on AccountStore is probably way more significant, but for the EVM work, we should just separate our code from the AccountStoreSC.tpp anyways. 

Bottom line: probably not worth it.

**Comment by valeryz** [comment]
Update: this is now rebased to `release/8.2`

**Review by chetan-zilliqa — APPROVED** [review]

**Review by yaron-zilliqa — APPROVED** [review]
